pkg/registry/apis/example/subresource.go

```diff
@@ -0,0 +1,57 @@
+package example
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	example "github.com/grafana/grafana/pkg/apis/example/v0alpha1"
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+	"github.com/grafana/grafana/pkg/services/grafana-apiserver/endpoints/request"
+)
+
+type DummySubresourceREST struct {
+	Store *genericregistry.Store
+}
+
+var _ = rest.Connecter(&DummySubresourceREST{})
+
+func (r *DummySubresourceREST) New() runtime.Object {
+	return &example.DummySubresource{}
+}
+
+func (r *DummySubresourceREST) Destroy() {
+}
+
+func (r *DummySubresourceREST) ConnectMethods() []string {
+	return []string{"GET"}
+}
+
+func (r *DummySubresourceREST) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, false, ""
```

Comment:
next we can try one that parses the query+path... but this is enough for this PR :)

Reply:
my memory is fuzzy, but i thought the path and query were available in the incoming request even if we don't parse them with a custom options type. is that correct?

Reply:
the false|true will specify if the path is registered or not -- when true the extra path object is added to openapi:
<img width="732" alt="image" src="https://github.com/grafana/grafana/assets/705951/6fa86052-f984-4c4b-8fe8-020f4dc03e36">

I have not figure out how (or if!) we can change the "path" name.

I *think* but have not yet confirmed, that the first option is to support automagic parsing the URL+parameters into an explicit object, and the third string parameter is the property where path should be shoved... but not yet confirmed :)